core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -639,6 +648,11 @@ public PlanNode plan(Update node)
                 // This column is updated...
                 Expression original = orderedColumnValues.get(index);
                 Expression setExpression = coerceIfNecessary(analysis, original, original);
+
+                // Rewrite check constraints based on update assignments
+                // The 'a > b' constraint with 'SET a = 1' assignment is rewritten as '1 > b'
+                constraints = Maps.transformValues(constraints, constraint -> replaceExpression(columnSchema.getName(), constraint, setExpression));
```

Comment:
This is not safe to do. For instance, if the update assignment is non-deterministic and the constraint references the column multiple times, the transformation is not equivalent.

The constraints should be planned as a filter on top of the subplan that computes the assignments, with the column references adjusted to refer to the outputs of such projection.

Reply:
@martint Updated. Please take another look.